### PR TITLE
Add Game_Config + CmdlineParser + video options - Config 2 of N

### PR DESCRIPTION
**Pull request by fmatthew5876**
The PR provides the following

- [x] A set of objects for manipulating configuration parameters, including the ability to disable them or lock them to specific values for platforms where it makes sense to do so.
- [x] A Game_Config class for storing EasyRPG configuration
- [x] Beginning support for config file parsing video options via `--config` `-c`
- [x] A new CmdlineParser class for command line arguments

### CODE REVIEW

**Comment by fmatthew5876** [comment]
I need to do another round of testing, but this is ready for review now.

**Comment by carstene1ns** [comment]
Question: Why do we need the new command line parser? Seems like inventing the wheel when there are ready to use alternatives, e.g. [CLI11](https://github.com/CLIUtils/CLI11), [Lyra](https://github.com/bfgroup/Lyra), [cxxopts](https://github.com/jarro2783/cxxopts).

**Comment by fmatthew5876** [comment]
We need a command line parser so it's easier to specify options and try to minimize the amount of work for exposing a config parameter between commandline / config file / scene settings.

Whether we use a third party one or write our own doesn't much matter to me. Command line parsing is a simple exercise. Writing one seemed like less work than evaluating all these other options and then getting them to compile on all our platforms.

This one has an unusual interface in that it parses and advances to the next in the same call. So it doesn't fit a traditional loop. This however is the most efficient way to do it and doesn't do any unnecessary extra parsing.

Whatever we choose. It needs to support unicode and case (in)sensitivity the way we use them. It also needs to be able to support the legacy RPG_RT command line options seamlessly combined with more modern ones.

File: src/config_param.h [hunk 1]
```diff
@@ -0,0 +1,377 @@
+/*
```
Comment (carstene1ns) [inline src/config_param.h:1]: mixed whitespace.

Comment (fmatthew5876) [reply src/config_param.h:1]: Fixed

File: src/game_config.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		video.vsync.Set(ini.GetBoolean("video", "show-fps", false));
+	}
+	if (ini.HasValue("video", "fps-render-window")) {
+		video.vsync.Set(ini.GetBoolean("video", "fps-render-window", false));
+	}
+	if (ini.HasValue("video", "fps-limit")) {
+		video.vsync.Set(ini.GetInteger("video", "fps-limit", 0));
+	}
+	if (ini.HasValue("video", "window-zoom")) {
+		video.vsync.Set(ini.GetInteger("video", "window-zoom", 0));
```
Comment (Ghabry) [inline src/game_config.cpp:137]: This all sets the wrong field ``video.vsync`` intentional because WIP (add comment :)) or a bug?

Comment (fmatthew5876) [reply src/game_config.cpp]: Thats a bug for sure. Fixed!

File: src/config_param.h [hunk 3]
```diff
@@ -172,189 +172,189 @@ using DoubleConfigParam = RangeConfigParam<double>;
...
+		lcf::FlagSet<E> _valid = ~lcf::FlagSet<E>();
+
+		E GetFirstValid() const {
+			for (size_t i = 0; i < _valid.size(); ++i) {
+				auto e = static_cast<E>(i);
+				if (_valid[e]) {
+					return e;
+				}
 			}
+			return E{};
```
Comment (Ghabry) [inline src/config_param.h:356]: the last commit added tab before public:/private:.  I think your editor is too smart with formating :)

File: src/game_config.cpp [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		of << "vsync=" << int(video.vsync.Get()) << "\n";
+	}
+	if (video.fullscreen.Enabled()) {
+		of << "fullscreen=" << int(video.fullscreen.Get()) << "\n";
+	}
+	if (video.show_fps.Enabled()) {
+		of << "show-fps=" << int(video.show_fps.Get()) << "\n";
+	}
+	if (video.fps_render_window.Enabled()) {
+		of << "show-fps=" << int(video.fps_render_window.Get()) << "\n";
```
Comment (Ghabry) [inline src/game_config.cpp:161]: must be ``fps-render-window``

**Pull request merged**